### PR TITLE
TOML Parser: Added 'time' support

### PR DESCRIPTION
**Pull request by sanket143**
Added ``time`` support in TOML Parser as stated in #7104

### CODE REVIEW

**Comment by sanket143** [comment]
@ben-albrecht Please review this pull request :yum:

File: test/library/packages/TOML/test/time.chpl [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+use TOML;
+
+config const str: string = """[owner]
```
Comment (ben-albrecht) [inline test/library/packages/TOML/test/time.chpl]: Could you add a test for a time with decimals? e.g. `timestamp2 = 00:32:00.999999`

File: modules/packages/TOML.chpl [hunk 2]
```diff
@@ -370,6 +371,16 @@ module TomlParser {
           Date = d;
           return Date;
         }
+        // Time
+        else if ti.match(val) {
+          var raw = getToken(source).split(":");
+          var t = new time(raw[1]: int,
+                           raw[2]: int,
+                           raw[3]: int);
```
Comment (ben-albrecht) [inline modules/packages/TOML.chpl]: Won't this be problematic for seconds with a decimal?

File: test/library/packages/TOML/test/time.chpl (hunk 1 above)
Comment (sanket143) [reply test/library/packages/TOML/test/time.chpl]: Actually I haven't added fractional second support. I'll add that too.. :+1:

Comment (sanket143) [reply test/library/packages/TOML/test/time.chpl]: Heya, I've made the changes. @ben-albrecht please review it.

File: modules/packages/TOML.chpl (hunk 2 above)
Comment (sanket143) [reply modules/packages/TOML.chpl]: :+1: Yeah.. I've added fractional seconds support.

**Comment by ben-albrecht** [comment]
Hey @sanket143 - I think this is close to merging. If you don't think you'll have time to finish it up, let me know, so we can close it for inactivity or make the changes to get it merged, if I find the time.

**Comment by sanket143** [comment]
Sorry. I will make the requested changes tonight.

File: modules/packages/TOML.chpl [hunk 3]
```diff
@@ -145,6 +145,7 @@ module TomlParser {
       dt = compile('^\\d{4}-\\d{2}-\\d{2}[ T]\\d{2}:\\d{2}:\\d{2}$'),
       realNum = compile("\\+\\d*\\.\\d+|\\-\\d*\\.\\d+|\\d*\\.\\d+"),
       ld = compile('^\\d{4}-\\d{2}-\\d{2}$'),
+      ti = compile('^\\d{2}:\\d{2}:\\d{2}(.\\d{6})?$'),
```
[the remark this answers is not in the document]
Comment (sanket143) [reply modules/packages/TOML.chpl]: @ben-albrecht I've made the changes, please review it.

File: modules/packages/TOML.chpl [hunk 4]
```diff
@@ -370,6 +371,34 @@ module TomlParser {
           Date = d;
           return Date;
         }
+        // Time
+        else if ti.match(val) {
+          var raw = getToken(source).split(":");
```
Comment (ben-albrecht) [inline modules/packages/TOML.chpl]: This works, but I'd like to suggest we use some built-in IO features to handle the rounding / edge cases, e.g.

```chpl
        else if ti.match(val) {
          var raw = getToken(source).split(":");
          var sec = '%.6dr'.format(raw[3]:real).split('.');

          var t: time;
          t = new time(raw[1]: int,
                       raw[2]: int,
                       sec[1]: int,
                       sec[2]: int);

          return t;
        }
```

File: test/library/packages/TOML/test/time.chpl [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+                            name = "Foo Bar"
+                            timestamp1 = 06:30:30
+                            timestamp2 = 06:30:30.123456
+                            timestamp3 = 06:30:30.1234567
+                            """;
```
Comment (ben-albrecht) [inline test/library/packages/TOML/test/time.chpl]: How about a couple other cases?

```chpl
config const str: string = """[owner]
                            name = "Foo Bar"
                            timestamp1 = 06:30:30
                            timestamp2 = 06:30:30.123456
                            timestamp3 = 06:30:30.1234567 # rounding up
                            timestamp4 = 06:30:30.1234564 # rounding down
                            timestamp5 = 06:30:30.9999999 # edge case
                            """;
```

File: modules/packages/TOML.chpl (hunk 4 above)
Comment (ben-albrecht) [reply modules/packages/TOML.chpl]: Note the formatted IO `%.6dr` should ensure there's always a second element in the `split('.')`

Comment (sanket143) [reply modules/packages/TOML.chpl]: well, didn't know that, this one's great actually.

**Comment by sanket143** [comment]
@ben-albrecht I've made the changes, actually even I think this is my best commit because even I was not so happy by handling microsecond that way, this one was great.
Plus my previous logic does had a bug. It was resolving ``06:30:30.9999999`` to ``06:30:30.999999`` which should be ``06:30:31`` which I realized after I used ``'%.6dr'.format(raw[3]:real)``

**Pull request merged**